### ISSUE TITLE
torchvision operators registered issue

### ISSUE THREAD

**Issue by swordnosword** [pytorch/vision#3833]
### In order to get the torchvision operators registered with torch (eg. for the JIT), I "#include <torchvision/vision.h>", but I got nothing after I run my project:
#include <iostream>
#include <torchvision/vision.h>
#include <torch/torch.h>
#include <torch/script.h>
#include <ops.h>

int main() {
    auto& ops = torch::jit::getAllOperators();
    std::cout << "torch jit operators\n";
    for (auto& op: ops) {
        auto& name = op->schema().name();
        if (name.find("torchvision") != std::string::npos)
            std::cout << "op : " << name << "\n";
    }
    std::cout << "\n";
}
### Enviroment
libtorch:1.8.1
torchvision:0.9.1
### CMakeLists.txt
cmake_minimum_required(VERSION 3.5)

project(deploydetection LANGUAGES CXX)

set(CMAKE_CXX_STANDARD 17)
set(CMAKE_CXX_STANDARD_REQUIRED ON)

set(Torch_DIR /home/appuser/libtorch/share/cmake/Torch)
find_package(Torch REQUIRED)

set(TORCHVISION_LIBS /home/appuser/vision/install/lib/libtorchvision.so)
include_directories(/home/appuser/vision/install/include /home/appuser/vision/install/include/torchvision/ops)

add_executable(deploydetection main.cpp)
target_link_libraries(deploydetection "${TORCH_LIBRARIES}")
target_link_libraries(deploydetection ${TORCHVISION_LIBS})

### When I run your [test project](https://github.com/pytorch/vision/tree/master/test/tracing/frcnn), I got an error:
Other error: 
Unknown builtin op: torchvision::nms.
Could not find any similar ops to torchvision::nms. This op may not exist or may not be currently supported in TorchScript.
:
  File "/home/pc/anaconda3/lib/python3.6/site-packages/torchvision/ops/boxes.py", line 36
    """
    _assert_has_ops()
    return torch.ops.torchvision.nms(boxes, scores, iou_threshold)
           ~~~~~~~~~~~~~~~~~~~~~~~~~ <--- HERE
Serialized   File "code/__torch__/torchvision/ops/boxes.py", line 75
  _29 = __torch__.torchvision.extension._assert_has_ops
  _30 = _29()
  _31 = ops.torchvision.nms(boxes, scores, iou_threshold)
        ~~~~~~~~~~~~~~~~~~~ <--- HERE
  return _31
def _box_inter_union(boxes1: Tensor,
'nms' is being compiled since it was called from 'batched_nms'
  File "/home/pc/anaconda3/lib/python3.6/site-packages/torchvision/ops/boxes.py", line 75
        offsets = idxs.to(boxes) * (max_coordinate + torch.tensor(1).to(boxes))
        boxes_for_nms = boxes + offsets[:, None]
        keep = nms(boxes_for_nms, scores, iou_threshold)
               ~~~ <--- HERE
        return keep
Serialized   File "code/__torch__/torchvision/ops/boxes.py", line 50
    _18 = torch.slice(offsets, 0, 0, 9223372036854775807, 1)
    boxes_for_nms = torch.add(boxes, torch.unsqueeze(_18, 1), alpha=1)
    keep = __torch__.torchvision.ops.boxes.nms(boxes_for_nms, scores, iou_threshold, )
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ <--- HERE
    _11 = keep
  return _11
'batched_nms' is being compiled since it was called from 'RegionProposalNetwork.filter_proposals'
Serialized   File "code/__torch__/torchvision/models/detection/rpn.py", line 66
    _11 = __torch__.torchvision.ops.boxes.clip_boxes_to_image
    _12 = __torch__.torchvision.ops.boxes.remove_small_boxes
    _13 = __torch__.torchvision.ops.boxes.batched_nms
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ <--- HERE
    num_images = (torch.size(proposals))[0]
    device = ops.prim.device(proposals)
'RegionProposalNetwork.filter_proposals' is being compiled since it was called from 'RegionProposalNetwork.forward'
  File "/home/pc/anaconda3/lib/python3.6/site-packages/torchvision/models/detection/rpn.py", line 357
        proposals = self.box_coder.decode(pred_bbox_deltas.detach(), anchors)
        proposals = proposals.view(num_images, -1, 4)
        boxes, scores = self.filter_proposals(proposals, objectness, images.image_sizes, num_anchors_per_level)
                        ~~~~~~~~~~~~~~~~~~~~~ <--- HERE
    
        losses = {}
Serialized   File "code/__torch__/torchvision/models/detection/rpn.py", line 39
    proposals = (self.box_coder).decode(torch.detach(pred_bbox_deltas0), anchors, )
    proposals0 = torch.view(proposals, [num_images, -1, 4])
    _8 = (self).filter_proposals(proposals0, objectness0, images.image_sizes, num_anchors_per_level, )
                                                                              ~~~~~~~~~~~~~~~~~~~~~ <--- HERE
    boxes, scores, = _8
    losses = annotate(Dict[str, Tensor], {})
### Enviroment
python:both 3.6.5 and 3.8.8
pytorch:1.8.1 torchvision:0.9.1
c++: libtorch:1.8.1 vision: 0.9.1

**Thanks!**

**Comment by datumbox** [comment]
Could you please try including `#include <torchvision/ops/nms.h>` and let me know whether you still get an issue?

**Comment by swordnosword** [comment]
Yes, I still get the same issue after including `#include <torchvision/ops/nms.h>`. Actually, I run your [test project](https://github.com/pytorch/vision/tree/master/test/tracing/frcnn) which has already  including `#include <torchvision/ops/nms.h>`.

**Comment by datumbox** [comment]
We have faced similar ops registration issues in the past but that was on Windows (see #3380). Could you also add the following right after the includes and let me know if it fixes the problem?

`static auto _nms = &vision::ops::nms;`

@bmanga Any thoughts about this?

**Comment by swordnosword** [comment]
Yes, I add the code. But I still get the same issue. By the way, my operating system is Ubuntu 18.04.

**Comment by swordnosword** [comment]
I see. I just change `set(CMAKE_CXX_STANDARD 17)` into`set(CMAKE_CXX_STANDARD 14)`, and run the test project successfully. Once I finish my custom project, I will close this issue. Thank you! @datumbox

**Comment by bmanga** [comment]
@datumbox It seems strange that changing to c++14 fixes things, but I will have a look this week. We should probably just do the linker vodoo in cmake like pytorch currently does.
